docs/preset-env.md

```diff
@@ -229,38 +229,64 @@ This option is useful for "blacklisting" a transform like `@babel/plugin-transfo
+
+```sh
+npm install core-js@3 --save
+```
+
 #### `useBuiltIns: 'entry'`
 
-> NOTE: Only use `require("@babel/polyfill");` once in your whole app.
-> Multiple imports or requires of `@babel/polyfill` will throw an error since it can cause global collisions and other issues that are hard to trace.
-> We recommend creating a single entry file that only contains the `require` statement.
+> NOTE: Only use `import "core-js";` and `import "regenerator-runtime/runtime";` once in your whole app.
+> Multiple imports or requires of those packages might cause global collisions and other issues that are hard to trace.
+> We recommend creating a single entry file that only contains the `import` statement.
 
-This option enables a new plugin that replaces the statement `import "@babel/polyfill"` or `require("@babel/polyfill")` with individual requires for `@babel/polyfill` based on environment.
+This option enables a new plugin that replaces the `import "core-js";` and `import "regenerator-runtime/runtime"` statements (or `require("corejs")` and `require("regenerator-runtime/runtime")`) with individual requires to different `core-js` entry points based on environment.
 
-```sh
-npm install @babel/polyfill --save
+**In**
+
+```js
+import "core-js";
 ```
 
+**Out (different based on environment)**
+
+```js
+import "core-js/modules/es.string.pad-start";
+import "core-js/modules/es.string.pad-end";
+```
+
+Importing `"core-js"` loads polyfills for every possible ECMAScript feature: what if you know that you only need some of them? When using `core-js@3`, `@babel/preset-env` is able to optimize every single `core-js` entrypoint and their combinations. For example, you might want to only polyfill array methods and new `Math` proposals:
+
 **In**
 
 ```js
-import "@babel/polyfill";
+import "core-js/es/array";
+import "core-js/proposals/math-extensions";
 ```
 
 **Out (different based on environment)**
 
 ```js
-import "core-js/modules/es7.string.pad-start";
-import "core-js/modules/es7.string.pad-end";
+import "core-js/modules/es.array.unscopables.flat";
+import "core-js/modules/es.array.unscopables.flat-map";
+import "core-js/modules/esnext.math.clamp";
+import "core-js/modules/esnext.math.deg-per-rad";
+import "core-js/modules/esnext.math.degrees";
+import "core-js/modules/esnext.math.fscale";
+import "core-js/modules/esnext.math.rad-per-deg";
+import "core-js/modules/esnext.math.radians";
+import "core-js/modules/esnext.math.scale";
 ```
 
-This will also work for `core-js` directly (`import "core-js";` or `require('core-js');`)
+> NOTE: When using `core-js@2` (either explicitly using the [`corejs: 2`](#corejs) option or implicitly), `@babel/preset-env` will also imports and requires of `@babel/polyfill`.
+> This behavior is deprecated because it isn't possible to use `@babel/polyfill` with different `core-js` versions.
 
-#### `useBuiltIns: 'usage'` (experimental)
+#### `useBuiltIns: 'usage'`
```

Comment:
As a sidenote, if we're going to remove the experimental tag here, we should enumerate all the caveats of using `usage` (e.g. `includes` doesn't work, etc) as there's lots of issues/confusion here.

Reply:
Well, `includes` will work after the core-js 3 pr

Reply:
Ah, I must have missed that.

Does it fix it for core-js 2 (didn't see that mentioned)? If not, it's even more reason for us to enumerate the caveats depending on the version of core-js being used?

Reply:
I think that we both confused `@babel/preset-env` with `@babel/transform-runtime` :stuck_out_tongue: . Instance methods were already polyfilled when using `core-js@2` [repl](https://babeljs.io/repl/build/master#?babili=false&browsers=&build=&builtIns=usage&spec=false&loose=false&code_lz=GYexDoEsDsGMBsCuATApgZwNxA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env&prettier=false&targets=&version=7.3.4)

Input:
```js
foo.includes;
```

Output:
```js
require("core-js/modules/es7.array.includes");
require("core-js/modules/es6.string.includes");

foo.includes;
```

Reply:
@nicolo-ribaudo with `core-js@2`, `useBuiltIns: usage` had really many limitations and bugs. With `core-js@3` target a serious part of them was fixed and I think that we can remove `experimental` tag.

Reply:
Just some examples:
```js
new Map(); // => '[object Object] - `Object#toString` definition was broken
Object.getOwnPropertySymbols; // => undefined - this definition also was broken

import(something).then(...); // => Error  - `Promise` is not polyfilled in Webpack helpers

global.Set; // => undefined

something.finally;

Promise.resolve().finally; // => most likely undefined because of incorrect polyfilling order

'values' in Object; // => false
```
and many other.

Reply:
Sure, it's awesome that `usage` with `core-js@3` is much improved over the previous version, but we need to communicate to users caveats and issues with with both versions as not everyone may have the freedom or luxury to upgrade immediately.

@nicolo-ribaudo I was referring to the the `@babel/preset-env` `includes` option not working with `usage`